src/components/TypeLayout/__tests__/TypeLayout.spec.js

```diff
@@ -3,53 +3,80 @@
+import {
+  TypeLayout,
+  TypeLayoutBody,
+  TypeLayoutRow,
+  TypeLayoutCell,
+} from '../../..';
 
 describe('TypeLayout', () => {
-  let typeLayout;
-
-  beforeEach(() => {
-    typeLayout = shallow(
+  test('should have no Axe or DAP violations', async () => {
+    const main = document.createElement('main');
+    render(
       <TypeLayout>
         <TypeLayoutBody>
-          {rows.map(row => {
-            const { id, title, description } = row;
+          <TypeLayoutRow>
+            <TypeLayoutCell>test cell title 1</TypeLayoutCell>
+            <TypeLayoutCell>test cell content 1</TypeLayoutCell>
+          </TypeLayoutRow>
+          <TypeLayoutRow>
+            <TypeLayoutCell>test cell title 2</TypeLayoutCell>
+            <TypeLayoutCell>test cell content 2</TypeLayoutCell>
+          </TypeLayoutRow>
+        </TypeLayoutBody>
+      </TypeLayout>,
+      {
+        // DAP requires a landmark '<main>' in the DOM:
+        container: document.body.appendChild(main),
+      }
+    );
+    await expect(document.body).toHaveNoAxeViolations();
+    await expect(document.body).toHaveNoDAPViolations('TypeLayout');
+  });
 
-            return (
-              <TypeLayoutRow key={id}>
-                <TypeLayoutCell>{title}</TypeLayoutCell>
-                <TypeLayoutCell>
-                  <ul>
-                    <li>{description}</li>
-                    <li>{description}</li>
-                    <li>{description}</li>
-                  </ul>
-                </TypeLayoutCell>
-              </TypeLayoutRow>
-            );
-          })}
+  test('should add a custom class to each component', () => {
+    const { getByText } = render(
+      <TypeLayout className="custom-layout-class">
+        <TypeLayoutBody className="custom-body-class">
+          <TypeLayoutRow className="custom-row-class">
+            <TypeLayoutCell className="custom-cell-class">
+              test cell
+            </TypeLayoutCell>
+          </TypeLayoutRow>
         </TypeLayoutBody>
       </TypeLayout>
     );
+    const cell = getByText(/test cell/i);
+    expect(cell).toHaveClass('custom-cell-class');
+    // Row class:
+    expect(cell.parentNode).toHaveClass('custom-row-class');
+    // Body class:
+    expect(cell.parentNode.parentNode).toHaveClass('custom-body-class');
+    // Layout wrapper class:
+    expect(cell.parentNode.parentNode.parentNode).toHaveClass(
+      'custom-layout-class'
+    );
   });
 
-  describe('render', () => {
-    it('renders correctly', () => {
-      expect(typeLayout).toMatchSnapshot();
-    });
-
-    it("renders the HTML of the node's subtree", () => {
-      expect(typeLayout.render()).toMatchSnapshot();
-    });
-
-    it('renders correct classes if `size` or `bordered` are set', () => {
-      typeLayout.setProps({ size: 'sm', border: true });
-      expect(typeLayout.render()).toMatchSnapshot();
-    });
+  test('should pass through extra props via spread attribute', () => {
```

Comment:
This is a reminder to also include tests for `border`, `children`, and `size`

Reply:
`border` is a Carbon `StructuredList` prop.
All of those props, particularly `children`, are demonstrated in the Storybook 🤔

Reply:
On initial view `border` and `size` look like a classes we add ourselves based on the prop, so my assumption is we'd want to cover that.

Also, so I fully understand, do you suggest we should avoid unit testing these props because they're covered by the examples in Storybook?

Reply:
Well so yes and no. 🤔 The snapshots were not helpful in this regard (they aren't checked visually for any meaning; they are just updated when they need to be updated). I can check for the presence of class names but that doesn't seem in the right spirit & the test itself seems less valuable than seeing what is rendered in storybook.